jsapp/js/account/security/email/emailSection.component.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {useEffect, useState} from 'react';
+import sessionStore from 'js/stores/session';
+import {EmailResponse, getUserEmails, setUserEmail} from './emailSection.api';
+import style from './emailSection.module.scss';
+import Button from 'jsapp/js/components/common/button';
+import TextBox from 'jsapp/js/components/common/textBox';
+import Icon from 'jsapp/js/components/common/icon';
+
+interface EmailState {
+  isLoading: boolean;
+  emails: EmailResponse[];
+  newEmail: string;
+}
+
+export default function EmailSection() {
+  const [session] = useState(() => sessionStore);
+
+  const [email, setEmail] = useState<EmailState>({
+    isLoading: true,
+    emails: [],
+    newEmail: '',
+  });
+
+  useEffect(() => {
+    if (email.isLoading) {
```

Comment:
Maybe naming it `!email.isInitialised` would make more sense here?

Reply:
We should standardize the language. I take "isLoading" to mean it's waiting for some async fetch of data. isInitialised I would interpret as the first load has completed.  Or perhaps more generically it just means the initial value was set. isLoading may flip back and forth, while isInitialised would load flip from false to true exactly once. isInitialised is helpful for things like a "app shell" UI that shouldn't show again later.

But this effect's design needs improved. It's being used as if it was what I describe as isInitialised. Omitting the second param of useEffect causes it to run every time (unnecessary). It works because isLoading starts true. This is untrue. The API data is not loading before the component even mounted. I think the intention here was to run once on mount. Therefore it should have a second param of `[]` and no additional if statements.

Let's keep the language conversation going. I would personally submit two booleans:

- isLoading
- isFirstLoadComplete

To me this is unambiguous.

Reply:
In [`4fe338a05`](https://github.com/kobotoolbox/kpi/pull/4163/commits/4fe338a050f0fb173c46952a55c02889d0000c8f), I removed isLoading from the state and passed `[]` to useEffect.